### ISSUE TITLE
Feature: Clicking the systray icon should minimize/close Ferdium to sytray

### ISSUE THREAD

**Issue by ppvnf** [ferdium/ferdium-app#675]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/ferdium/ferdium-app/blob/develop/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/ferdium/ferdium-app/blob/develop/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/ferdium/ferdium-app/issues) for a feature request that matches the one I want to file, without success.

### Problem Description

In the taskbar you can click the Ferdium icon to minimize or restore Ferdium window, however the same functionality isn't available for the system tray. While you can click the tray icon to restore Ferdium, it's not possible to click it to minimize/close to the systray.

For example: I have the option close to systray enabled, so when I want to quickly check some message I click the Ferdium icon in the system tray, however clicking the Ferdium icon again (after checking the message) won't close/minimize it to the systray (it's necessary to click the X button instead).

### Proposed Solution

Clicking the systray icon should minimize/close Ferdium to sytray if Ferdium isn't already closed/minimized to the systray

### Alternatives Considered

Maybe have this as an option, but I think this feature is pretty standard tbh

**Comment by vraravam** [comment]
I'm running the latest nightly, on macos, and I see that the expected behavior from the OP is how Ferdium's systray icon is behaving. Closing the card. Please reopen if this is still not fixed.

**Comment by tonypachino** [comment]
This doesn't work in Linux KDE, can you please fix that, so that i can one click to maximize an minimize window?

**Comment by ppvnf** [comment]
@vraravam Could you reopen it? It's not fixed on Windows 10 (and KDE apparently)

**Comment by SinTan1729** [comment]
To add to this, it used to work for me (on KDE) about a month ago, but suddenly stopped working. Unfortunately, I'm not sure if a KDE update or a Ferdium update broke it though. But maybe this might help narrow the issue down.

**Comment by Alphrag** [comment]
Hi all, just had a look at that and I think the problem is coming from the fact that we only minimise if the app window is currently on focus. This means for example that if your Ferdium tray icon is directly onto the bar and your app is currently on focus, clicking on the tray icon minimises it. 
However, if it is located on an overflow menu (which is the default on Windows for example), then the simple fact that you have to open this makes the Ferdium window no longer on focus, and thus clicking the tray icon brings it back onto focus. This means in particular, that if it was already your top window, it seems that nothing has happened, when it has in fact lost focus and gained it again. So this is dependent not only on how the OS deals with the focus of multiple elements and windows, but also on your own choice on tray disposition.

Changing this behaviour would break the normal use of these that have a direct access to the tray icon (especially on Mac) but I could see that a setting forcing the minimisation if Ferdium is open (wherever that window might be) when the tray icon is clicked could be possible. 
@vraravam What would you think of that?

**Comment by SinTan1729** [comment]
@Alphrag I don't think this is correct. My tray icon is directly on the bar, but it still doesn't work.
![Screen recording](https://user-images.githubusercontent.com/35728385/208337241-141c5bf4-96a5-4b33-8d57-f1308fafc10d.mp4)

**Comment by Alphrag** [comment]
> @Alphrag I don't think this is correct. My tray icon is directly on the bar, but it still doesn't work. 

@SinTan1729 I don't see the problem in your recording, the show/hide Ferdium in the tray menu seems to work perfectly fine. Is it the single click that causes problem?

**Comment by SinTan1729** [comment]
> Is it the single click that causes problem?

@Alphrag It used to be the case that only one click was needed to maximize/minimize the app. I just needed to click on the tray icon once, and not deal with the drop-down at all.

**Comment by Alphrag** [comment]
Ok, then it seems to be a problem between KDE and electron, since we handle the left-click on the tray icon in the same way we do for the show/hide element in the tray menu.  I have no problem on my fedora with gnome (and don't recall having it on my debian and ubuntu, even though it is a feature I rarely use).
On Windows the value and behaviour returned by `WebBrowser.isFocused()` is quite inconsistent which makes the result of clicking on the tray icon unknown. A similar thing could happen with KDE. 
I will see if there is a good way to look at the taskbar to infer if the window in "on top", but I'm afraid we won't have the possibility to have the full feature if we do so (as in: if the window is "visible" on any desktop then clicking will hide it instead of putting it into focus).

**Comment by ppvnf** [comment]
[Rambox ](https://github.com/ramboxapp/community-edition) used to allow double-clicking the systray icon to allow hiding/showing the window. Sorry I've not much programming knowledge to help, but since both are open-source and based on Electron, couldn't this feature be ported to Ferdium? (although I do think single-clicking would be preferable)

<details>

https://user-images.githubusercontent.com/73800734/209119588-eaaa74f6-20a1-4894-aa1b-97db27772f10.mp4

</details>

**Comment by 392781** [comment]
I would like to add that currently, minimization works but when you bring Feridum back by clicking on the app icon, it opens a new window for a brief second before closing it.
